### DISCUSSION TITLE
adding-decorated-classes-to-the-asp.net-core-di-container-using-scrutor/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#60] [Announcements]
# adding-decorated-classes-to-the-asp.net-core-di-container-using-scrutor/

In this post I show how to do service decoration with Scrutor, where services "wrap" another implementation of the service

https://andrewlock.net/adding-decorated-classes-to-the-asp.net-core-di-container-using-scrutor/

**Comment by tselofan** [comment]
Hi. Important note. If your decorated service implements disposable, it wont dispose in the end of lifetime. Only the decorator. Right?

[a comment by an automated account is not shown]

**Comment by pwelter34** [comment]
Hello,  You might consider using the Injectio project.  Injectio is very similar to Scrutor except it uses source generator to do the registration at compile time.

https://github.com/loresoft/Injectio

thanks,
~ Paul
